.github/workflows/nightly_release.yml

```diff
@@ -0,0 +1,108 @@
+name: SalesforceCommerceCloud/pwa-kit/nightly_release
+on:
+  workflow_dispatch:
+  schedule:
+    # Run weekdays at 12am (PST) - cron uses UTC times
+    - cron: "0 8 * * 1-5"
+
+jobs:
+  create_nightly_release:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Get Timestamp
+        run: |-
+          echo "release_timestamp=$(date +'%Y%m%d%H%M%S')" >> "$GITHUB_ENV"
+
+      - name: Get monorepo version
+        run: |-
+          version=`jq -r ".version" package.json | cut -d "-" -f 1`
+          echo "monorepo_version_base=$version" >> "$GITHUB_ENV"
+          version_major=`echo "$version" | cut -d "." -f 1`
+          echo "monorepo_version_major=$version_major" >> "$GITHUB_ENV"
+          echo "nightly_version=${{ env.monorepo_version_base }}-nightly-${{ env.release_timestamp }}" ?? "$GITHUB_ENV"
+
+      - name: Get retail-react-app version
+        run: |-
+          version=`jq -r ".version" packages/template-retail-react-app/package.json | cut -d "-" -f 1`
+          echo "retail_app_version_base=$version" >> "$GITHUB_ENV"
+
+      - name: Get commerce-sdk-react version
+        run: |-
+          version=`jq -r ".version" packages/commerce-sdk-react/package.json | cut -d "-" -f 1`
+          echo "commerce_sdk_react_version_base=$version" >> "$GITHUB_ENV"
+
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18
+          cache: npm
+
+      - name: Install Monorepo Dependencies
+        run: |-
+          # Install node dependencies
+          node ./scripts/gtime.js monorepo_install npm ci
+
+      - name: Create Release Branch
+        run: git switch release-nightly-${{ env.monorepo_version_major }}.x || git switch -c release-nightly-${{ env.monorepo_version_major }}.x
+
+        # Pull all changes from develop and keep changes from develop in case of merge-conflicts
+      - name: Pull latest changes from develop
+        run: |-
+          git switch develop
+          git merge -s ours release-nightly-${{ env.monorepo_version_major }}.x
+          git switch release-nightly-${{ env.monorepo_version_major }}.x
+          git merge develop
+          git switch release-nightly-${{ env.monorepo_version_major }}.x
```

Comment:
```suggestion
```
You're already on the branch! (Not a blocker because it doesn't error, just prints a warning.)